components/Auth/AuthButton.tsx

```diff
@@ -0,0 +1,8 @@
+import { getUserCredentials } from "lib/api/auth";
+import { LogoutButton } from "./LogoutButton";
+import { SigninButton } from "./SigninButton";
+
+export default async function AuthButton() {
+  const user = await getUserCredentials();
+  return user ? <LogoutButton /> : <SigninButton />;
```

Comment:
That is only happening because the Github OAuth has the homepage url set as verto.sh. That's why it will only work in the main production branch having the same Url. You can create another github oauth app for testing.